.github/workflows/codeql.yaml

```
@@ -0,0 +1,43 @@
+---
+name: codeQL
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+jobs:
+  analyze:
+    name: Static analysis with CodeQL
```

For dependabot support when the bot will create PRs
```suggestion
    name: Static analysis with CodeQL
    permissions:
      actions: read
      contents: read
      security-events: write
```